docs/source/macos/nodejs.rst

```diff
@@ -1,115 +1,184 @@
-of the version of Node.js that is currently active.
+You can also add this to your alias list, so it is simpler to use (``nvmuse``
+for example).
 
-5 Switch between different versions of Node.js
-----------------------------------------------
 
-To switch between different versions of Node.js, execute on the command
-line:
+4 Switch between different versions of Node.js automatically
+------------------------------------------------------------
+
+If you wish to have ``nvm`` automatically switch to different Node.js version
+upon navigating inside the directory which contains ``.nvmrc`` file, you can
+configure the function for this in the ``~/.cdnvm`` file:
+
+.. code:: console
+
+    cdnvm() {
+        command cd "$@";
+        nvm_path=$(nvm_find_up .nvmrc | tr -d '\n')
+
+        # If there are no .nvmrc file, use the default nvm version
+        if [[ ! $nvm_path = *[^[:space:]]* ]]; then
+
+            declare default_version;
+            default_version=$(nvm version default);
+
+            # If there is no default version, set it to `node`
+            # This will use the latest version on your machine
+            if [[ $default_version == "N/A" ]]; then
+                nvm alias default node;
+                default_version=$(nvm version default);
+            fi
+
+            # If the current version is not the default version, set it to use the default version
+            if [[ $(nvm current) != "$default_version" ]]; then
+                nvm use default;
+            fi
+
+        elif [[ -s $nvm_path/.nvmrc && -r $nvm_path/.nvmrc ]]; then
+            declare nvm_version
+            nvm_version=$(<"$nvm_path"/.nvmrc)
+
+            declare locally_resolved_nvm_version
+            # `nvm ls` will check all locally-available versions
+            # If there are multiple matching versions, take the latest one
+            # Remove the `->` and `*` characters and spaces
+            # `locally_resolved_nvm_version` will be `N/A` if no local versions are found
+            locally_resolved_nvm_version=$(nvm ls --no-colors "$nvm_version" | tail -1 | tr -d '\->*' | tr -d '[:space:]')
+
+            # If it is not already installed, install it
+            # `nvm install` will implicitly use the newly-installed version
+            if [[ "$locally_resolved_nvm_version" == "N/A" ]]; then
+                nvm install "$nvm_version";
+            elif [[ $(nvm current) != "$locally_resolved_nvm_version" ]]; then
+                nvm use "$nvm_version";
+            fi
+        fi
+    }
+    alias cd='cdnvm'
+    cd "$PWD"
+
+and add this to your ``~/.bashrc`` file:
 
 .. code:: console
 
-   n
+   if [ -f ~/.cdnvm ]; then
+      . ~/.cdnvm # This enables automatic switch of nvm on folder change
+   fi
+
+
+5 Managing packages with ``nvm``
+--------------------------------
 
-Then select between available versions of Node.js.
+For some packages (``yarn`` usually), it is best to let ``nvm`` install it
```

Comment:
maybe a disclaimer that yarn doesn't work otherwise / has to be manually installed with `npm install -g yarn` with every new node version added to nvm? Since it's the recommended way instead of using apt/brew/ports.

Not sure where it might be best to add this, but I feel like this might be better earlier on because it's not really optional, maybe between sections 2 and 3? Or after 3?

Reply:
If a version's been installed before the default-packages file has been added, there are two options
1. add the file and run `nvm install <version>` again
2. just run `npm install -g yarn` while that version is active